### DISCUSSION TITLE
HTTPS error 12030

### DISCUSSION THREAD

**Discussion by AgrYpn1a** [scottoffen/grapevine#120] [Q&A] [answered]
Hey, I have an app that is sending http/s requests to my server, this app is already a binary, so I am unable to look at the source code, but I was provided with it along with the certificates for the secured connection.

I followed the docs on your website, and basically now I have this piece of code

```csharp
    _server = RestServerBuilder.UseDefaults().Build();
    _server.AfterStarting += Server_AfterStarting;
    _server.AfterStopping += Server_AfterStopping;

    _server.UseCorsPolicy();
    

    _server.Prefixes.Clear();
    _server.Prefixes.Add("https://*:6060/");
    _server.Prefixes.Add("https://*:443/");
```

but the app is unable to communicate with the server, it fails with the 12030 error. Do I need to add anything else to the configuration? I have been provided with `rootCA.pem`, `server.crt` and `server.key` files. I am on Windows, and have installed the `server.crt` certificate under trusted root certificates.

![image](https://user-images.githubusercontent.com/8715031/159433274-898077a8-9647-4021-8b7a-eb39100a919d.png)

Also I was unable to bind certificate to the port for some reason.
![image](https://user-images.githubusercontent.com/8715031/159434658-b4fb0bd2-c540-40cc-8c4b-f5b8910894e9.png)

And sometimes the other error
![image](https://user-images.githubusercontent.com/8715031/159436888-24517870-e984-4775-8423-06135962b529.png)

===EDIT 1
I finally managed to install the certificate by running ` winpty openssl pkcs12 -export -out ./server.pfx -inkey ./server.key -in ./server.crt` from the git shell on Windows and bind it to a port. However, still getting the same error from the app.

**Comment by scottoffen** [comment]
- What do you see in the logs?
- Do you get a response if you use something like Postman to test your server routes?

**Reply by AgrYpn1a** [reply]
Without https, from insomnia the url is `http://localhost:6060`
![image](https://user-images.githubusercontent.com/8715031/159674464-1146a628-1ae8-42c0-8a37-e813864afa97.png)

And now for https
![image](https://user-images.githubusercontent.com/8715031/159674657-21f394fa-c083-4f6e-8d2b-4bdafabb19b4.png)

I am unable to see the log, since I am running the server from Unity, I am guessing I need to pass it the custom logger to log messages via something else, but that's a separate topic. I opened a [discussion](https://github.com/scottoffen/grapevine/discussions/121) for creating a custom logger.

**Reply by AgrYpn1a** [reply]
@scottoffen I think I managed to redirect the logs. It seems that when I run the server on https the request isn't even reaching it, since there's no log.

**Comment by AgrYpn1a** [comment, marked as answer]
To wrap up the discussion: The issue is that Grapevine relies internally on HttpListener and I am using it in Mono implementation of .NET (in Unity3D) where there are some issues with the SSL. This renders Grapevine unusable in my situation, which is not Grapevines fault, thus I'll mark this discussion as answered.

@scottoffen btw, can you still check my question regarding the custom logger implementation, I'd really like an example as I may use Grapevine in my other projects, thanks a lot for your help!
